package.json

```diff
@@ -174,22 +174,22 @@
             },
             {
                 "key": "ctrl+enter",
-                "when": "editorTextFocus && inputFocus && notebookEditorFocused && notebookType == jupyter-notebook && config.jupyter.enableKeyboardShortcuts",
+                "when": "editorTextFocus && inputFocus && notebookEditorFocused && notebookType == jupyter-notebook && config.jupyter.enableKeyboardShortcuts || notebookCellListFocused && notebookCellType == 'code' && notebookType == jupyter-notebook && config.jupyter.enableKeyboardShortcuts",
```

Comment:
What's the priority for an || statement? There's no parenthesis here so not sure if the or applies to just the two items next to it or the whole statement after?

Reply:
I don't believe parentheses are supported in when clauses. At least I tried the following and it didn't work:

```
"when": "((editorTextFocus && inputFocus && notebookEditorFocused) || notebookCellListFocused && notebookCellType == 'code')) && notebookType = jupyter-notebook && config.jupyter.enableKeyboardShortcuts"
```

The priority is not documented, but I would be very surprised if && didn't trump || in precedence.

Reply:
I believe this is the spot for parsing when expressions. Looks like VS Code splits on '||', then '&&', so unless I'm misreading and this code isn't the right spot, it seems like our when clauses should do what we expect https://github.com/microsoft/vscode/blob/5b73c885890e95af601df0f968194efeab1b70cd/src/vs/platform/contextkey/common/contextkey.ts#L128-L144